CHANGES.md

```diff
@@ -12,21 +12,107 @@ To be released.
 
 ### Backward-incompatible API changes
 
+ -  Constructors of `Currency` class now take `(BigInteger Major),
+    BigInteger Minor)?` as the maximum supply parameter, and a boolean value to
+    mark whether the `Currency` is trackable.  [[#915], [#2200]]
+    -  *NOTE:* if you already have defined some `Currency` instances in prior
+       to the addition of maximum supply, and if it is on a live chain, you
+       cannot modify the already-defined `Currency` instances to have maximum
+       supply caps since it would make those `Currency` instances to be
+       effectively different ones compared to the original ones, as it would
+       change their `Hash`es. You must pass a `null` as their maximum supply
+       parameter. Also, you must set the `totalSupplyTrackable` parameter to
+       false.
```

Comment:
Any reason why  this should be *explicitly* set to `false`? Is there a case where we don't want to track but have total supply capped for new `Currency`?

Reply:
Actually, three configurations are possible:

- `maximumSupply is null` and `totalSupplyTrackable = false`, in which case no maximum supply cap nor total supply tracking is available;
- `maximumSupply is null` and `totalSupplyTrackable = true`, in which case no maximum supply cap is present but total supply tracking is available;
- `maximumSupply is { }` and `totalSupplyTrackable = true`, in which case maximum supply cap is set in place.

In case of `maximumSupply is { }` and `totalSupplyTrackable = false`, the constructor throws an [`ArgumentException`](https://github.com/planetarium/libplanet/pull/2200/commits/beb3e7b064fe36d7cf4419dcfa22c7e595404164#diff-4ba5d5e78e585f64e217441ed95e634f2d1b5c5df16ac3d419834d040882d7c4R103-R109).

Reply:
So it is for backward compatibility reason. Now I understand the context more. Then again, I'm wondering if it'd be possible to discourage using `(null, false)`, i.e. legacy `Currency`, more strongly in general. 🙄

Reply:
One of the efforts currently in place to avoid having `totalSupplyTrackable = false` is having `totalSupplyTrackable = true` as the default parameter, but now that I think about it, how would it be having public constructor for "new" `Currency` only, and having a private constructor that creates the legacy `Currency`, and having a `[Obsolete] public static Currency.CreateNonTrackableCurrency()`? cc @dahlia